### DISCUSSION TITLE
Implement a way to read/write itemmeta without expensive cloning

### DISCUSSION THREAD

**Discussion by ghost** [PaperMC/Paper#10496] [Ideas]
**Is your feature request related to a problem? Please describe.**
Itemstack.getItemmeta() even when you only want something like displayname clones the entire meta and if the item is a written book, runs heavy regex on every page taking excess of 100ms in some cases

**Describe the solution you'd like**
perhaps getitemmeta(boolean snapshot) or a new interface called liveItemmeta etc. Something that wouldn't break plugins already using it this way. Perhaps _parts_ of the meta could be initialized when the for example getlore() methods are called etc

**Describe alternatives you've considered**
There isn't really any other solution other than using NMS to read NBT an item's NBT tags.

**Additional context**

**Comment by electronicboy** [comment]
ItemMeta is contractually a snapshot, there is no way to implement ItemMeta without cloning the entire existing data structure; This would need to be an entirely separate API for this specific purpose

**Comment by MiniDigger** [comment]
I mean, couldn't you say the same about block states? yet paper still allows you not take a snapshot

**Comment by electronicboy** [comment]
That API works fundamentally differently to how ItemMeta works
